service/worker/archiver/activities.go

```diff
@@ -123,9 +126,15 @@ func deleteHistoryActivity(ctx context.Context, request ArchiveRequest) (err err
 	if err == nil {
 		return nil
 	}
-	logger := tagLoggerWithHistoryRequest(tagLoggerWithActivityInfo(container.Logger, activity.GetInfo(ctx)), &request)
-	logger.Error("failed to delete history events", tag.Error(err))
-	if !common.IsPersistenceTransientError(err) {
+
+	if _, ok := err.(*serviceerror.WorkflowNotReady); !ok {
+		logger := tagLoggerWithHistoryRequest(tagLoggerWithActivityInfo(container.Logger, activity.GetInfo(ctx)), &request)
```

Comment:
Suppress logs for notReady error which is expected for deleteWorkflowExecution call. The error will still be logged if all retry fails.